### DISCUSSION TITLE
How to Avoid Task Delays with Asynchronous HTTP Requests?

### DISCUSSION THREAD

**Discussion by a-leonid** [quartznet/quartznet#2389] [Q&A]
I have 3 tasks. Each task takes 2 seconds to complete, but the trigger interval is set to 500 ms. This means that the 3rd task will hang until one of the tasks is completed.

Each task essentially performs the following request:
HttpResponseMessage response = await client.SendAsync(request, cancellationToken);

Is there a way to avoid task hanging and ensure that the thread pool picks up the next task immediately, without waiting for the current task to complete?

**Comment by lahma** [comment]
Probably not Quartz related, but are you looking something like this?

```c#
Task<HttpResponseMessage> response1Task = client.SendAsync(request1, cancellationToken);
Task<HttpResponseMessage> response2Task = client.SendAsync(request2, cancellationToken);
Task<HttpResponseMessage> response3Task = client.SendAsync(request3, cancellationToken);
await Task.WhenAll(response1Task, response2Task, response3Task);

var result1 = await response1Task;
var result2 = await response2Task;
var result3 = await response3Task;
```

Error handling etc omitted.

**Reply by a-leonid** [reply]
No, I already have 3 configured tasks. I decided to test how the program would work with just 2 threads. I wrote a simple API to introduce a longer delay.

It turns out that the scheduler:

Takes task 1 and starts executing it.
Takes task 2 and starts executing it.
However, because of the long wait for a response, task 3 is not executed, even though all threads are essentially free.

I can't use Task.Run() inside tasks because I don't need task overlap.
